tcping.go

```diff
@@ -637,24 +649,26 @@ func tcping(tcpStats *stats) {
 	IPAndPort := netip.AddrPortFrom(tcpStats.userInput.ip, tcpStats.userInput.port)
 
 	connStart := time.Now()
-	conn, err := net.DialTimeout("tcp", IPAndPort.String(), time.Second)
-	connEnd := time.Since(connStart)
-	rtt := nanoToMillisecond(connEnd.Nanoseconds())
+	// if timeout = 0, there is no timeout, see net.Dial
+	conn, err := net.DialTimeout("tcp", IPAndPort.String(), tcpStats.userInput.timeout)
+	connDuration := time.Since(connStart)
+	rtt := nanoToMillisecond(connDuration.Nanoseconds())
 
 	if err != nil {
-		tcpStats.handleConnError(connStart)
+		tcpStats.handleConnError(connStart, maxDuration(connDuration, tcpStats.intervalBetweenProbes))
 	} else {
-		tcpStats.handleConnSuccess(rtt, connStart)
+		tcpStats.handleConnSuccess(rtt, connStart, maxDuration(connDuration, tcpStats.intervalBetweenProbes))
```

Comment:
Let's extract this `maxDuration` outside this condition.

```go
elapsed :=  maxDuration(connDuration, tcpStats.intervalBetweenProbes)
```

Then we just pass it to our conditions